shell/src/main/java/tachyon/shell/command/AbstractTfsShellCommand.java

```diff
@@ -51,4 +56,39 @@ public boolean validateArgs(String... args) {
     }
     return valid;
   }
+
+  /**
+   * Gets the expected number of arguments of the command.
+   *
+   * @return the number of arguments
+   */
+  abstract int getNumOfArgs();
+
+  @Override
+  public CommandLine parseAndValidateArgs(String... args) {
+    final Options opts = new Options();
+    // Add R option for recursively.
+    // TODO(ifcharming): extend more options as needed.
+    final Option recursive = Option.builder("R")
+        .required(false)
+        .hasArg(false)
+        .desc("recusively")
+        .build();
+
+    opts.addOption(recursive);
+
+    final CommandLineParser parser = new DefaultParser();
+    CommandLine cmd;
+    try {
+      cmd = parser.parse(opts, args, true /* stopAtNonOption */);
```

Comment:
Can you add a TODO here to improve the error message when an unregistered option appears? Right now I think `chgrp -R group2 /testDir -nonOption` will say there should be 2 args, but there are 3.